README
NAME

    Dist::Zilla::Plugin::MinimumPerl - Detects the minimum version of Perl
    required for your dist

VERSION

      This document describes v1.005 of Dist::Zilla::Plugin::MinimumPerl - released October 31, 2014 as part of Dist-Zilla-Plugin-MinimumPerl.

DESCRIPTION

    This plugin uses Perl::MinimumVersion to automatically find the minimum
    version of Perl required for your dist and adds it to the prereqs.

            # In your dist.ini:
            [MinimumPerl]

    This plugin will search for files matching /\.(t|pl|pm)$/i in the lib/,
    inc/, and t/ directories. If you need it to scan a different directory
    and/or a different extension please let me know.

ATTRIBUTES

 perl

    Specify a version of perl required for the dist. Please specify it in a
    format that Build.PL/Makefile.PL understands! If this is specified,
    this module will not attempt to automatically detect the minimum
    version of Perl.

    The default is: undefined ( automatically detect it )

    Example: 5.008008

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Dist::Zilla

      * Dist::Zilla::Plugin::MinimumPerlFast

SUPPORT

 Perldoc

    You can find documentation for this module with the perldoc command.

      perldoc Dist::Zilla::Plugin::MinimumPerl

 Websites

    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

      * MetaCPAN

      A modern, open-source CPAN search engine, useful to view POD in HTML
      format.

      http://metacpan.org/release/Dist-Zilla-Plugin-MinimumPerl

      * Search CPAN

      The default CPAN search engine, useful to view POD in HTML format.

      http://search.cpan.org/dist/Dist-Zilla-Plugin-MinimumPerl

      * RT: CPAN's Bug Tracker

      The RT ( Request Tracker ) website is the default bug/issue tracking
      system for CPAN.

      http://rt.cpan.org/NoAuth/Bugs.html?Dist=Dist-Zilla-Plugin-MinimumPer
      l

      * AnnoCPAN

      The AnnoCPAN is a website that allows community annotations of Perl
      module documentation.

      http://annocpan.org/dist/Dist-Zilla-Plugin-MinimumPerl

      * CPAN Ratings

      The CPAN Ratings is a website that allows community ratings and
      reviews of Perl modules.

      http://cpanratings.perl.org/d/Dist-Zilla-Plugin-MinimumPerl

      * CPAN Forum

      The CPAN Forum is a web forum for discussing Perl modules.

      http://cpanforum.com/dist/Dist-Zilla-Plugin-MinimumPerl

      * CPANTS

      The CPANTS is a website that analyzes the Kwalitee ( code metrics )
      of a distribution.

      http://cpants.cpanauthors.org/dist/overview/Dist-Zilla-Plugin-Minimum
      Perl

      * CPAN Testers

      The CPAN Testers is a network of smokers who run automated tests on
      uploaded CPAN distributions.

      http://www.cpantesters.org/distro/D/Dist-Zilla-Plugin-MinimumPerl

      * CPAN Testers Matrix

      The CPAN Testers Matrix is a website that provides a visual overview
      of the test results for a distribution on various Perls/platforms.

      http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-MinimumPerl

      * CPAN Testers Dependencies

      The CPAN Testers Dependencies is a website that shows a chart of the
      test results of all dependencies for a distribution.

      http://deps.cpantesters.org/?module=Dist::Zilla::Plugin::MinimumPerl

 Email

    You can email the author of this module at APOCAL at cpan.org asking
    for help with any problems you have.

 Internet Relay Chat

    You can get live help by using IRC ( Internet Relay Chat ). If you
    don't know what IRC is, please read this excellent guide:
    http://en.wikipedia.org/wiki/Internet_Relay_Chat. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You
    can join those networks/channels and get help:

      * irc.perl.org

      You can connect to the server at 'irc.perl.org' and join this
      channel: #perl-help then talk to this person for help: Apocalypse.

      * irc.freenode.net

      You can connect to the server at 'irc.freenode.net' and join this
      channel: #perl then talk to this person for help: Apocal.

      * irc.efnet.org

      You can connect to the server at 'irc.efnet.org' and join this
      channel: #perl then talk to this person for help: Ap0cal.

 Bugs / Feature Requests

    Please report any bugs or feature requests by email to
    bug-dist-zilla-plugin-minimumperl at rt.cpan.org, or through the web
    interface at
    http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Dist-Zilla-Plugin-Minimu
    mPerl. You will be automatically notified of any progress on the
    request by the system.

 Source Code

    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    https://github.com/apocalypse/perl-dist-zilla-plugin-minimumperl

      git clone git://github.com/apocalypse/perl-dist-zilla-plugin-minimumperl.git

AUTHOR

    Apocalypse <APOCAL@cpan.org>

 CONTRIBUTORS

      * Nigel Gregoire <nigelg@airg.com>

      * Olivier Mengué <dolmen@cpan.org>

      * Pedro Melo <melo@simplicidade.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.

DISCLAIMER OF WARRANTY

    THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY
    APPLICABLE LAW. EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT
    HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT
    WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT
    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
    PARTICULAR PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE
    OF THE PROGRAM IS WITH YOU. SHOULD THE PROGRAM PROVE DEFECTIVE, YOU
    ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR
    CONVEYS THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
    INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
    ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT
    NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES
    SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO
    OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY
    HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.



Changes
Revision history for Dist::Zilla::Plugin::MinimumPerl

1.005 2014-10-31T17:29:36Z UTC
 - Reverted the removal of the filename check while scanning files -
   RT#99935 ( thanks Nigel! )

 [STATISTICS]
 - code churn: 2 files changed, 256 insertions(+)

1.004 2014-10-28T02:21:06Z UTC
 - Expanded the prereq specification so we now separate the configure,
   test, and runtime phases RT#61028 ( thanks DAGOLDEN! )
 - Skip parsing files with encoding == bytes RT#96071 ( thanks ETHER! )

1.003 2011-04-21T16:19:04Z UTC
 - Make sure that we scan only .t and .pl/pm files ( thanks RT#67355 and
   DOHERTY )

1.002 2011-04-01T04:25:20Z UTC
 - Re-release because v1.001 somehow didn't get on CPAN... /me smacks
   myself!
 - Minor POD tweaks

1.001 2010-12-08T02:54:32Z UTC
 - Converted to Dist::Zilla for the release process!
 - Updated to support Dist::Zilla v4
 - Use MooseX::Types::Perl to validate the perl attr
 - Now scans only the t/, lib/, and executable files for perl code ( thanks
   DAGOLDEN )

0.02 0000-00-00T00:00:00Z UTC
 - d0h, the logic for finding the minimum Perl version was reversed!

0.01 0000-00-00T00:00:00Z UTC
 - initial release



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.021.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28",
    "perl" => "5.006"
  },
  "dist_abstract" => "Detects the minimum version of Perl required for your dist",
  "dist_author" => [
    "Apocalypse <APOCAL\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-MinimumPerl",
  "dist_version" => "1.004",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::MinimumPerl",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::FileFinderUser" => "4.200006",
    "Dist::Zilla::Role::PrereqSource" => "5.006",
    "Moose" => "1.03",
    "Moose::Util::TypeConstraints" => "1.01",
    "MooseX::Types::Perl" => "0.101340",
    "Perl::MinimumVersion" => "1.26",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88",
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Test::More" => "0.88",
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
#
# This file is part of Dist-Zilla-Plugin-MinimumPerl
#
# This software is copyright (c) 2014 by Apocalypse.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use strict; use warnings;
# This Makefile.PL for Dist-Zilla-Plugin-MinimumPerl was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.014
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Detects the minimum version of Perl required for your dist",
  "AUTHOR" => "Apocalypse <APOCAL\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-MinimumPerl",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::MinimumPerl",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::FileFinderUser" => "4.200006",
    "Dist::Zilla::Role::PrereqSource" => "5.006",
    "Moose" => "1.03",
    "Moose::Util::TypeConstraints" => "1.01",
    "MooseX::Types::Perl" => "0.101340",
    "Perl::MinimumVersion" => "1.26",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.005",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Dist::Zilla::Role::FileFinderUser" => "4.200006",
  "Dist::Zilla::Role::PrereqSource" => "5.006",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build::Tiny" => "0.039",
  "Moose" => "1.03",
  "Moose::Util::TypeConstraints" => "1.01",
  "MooseX::Types::Perl" => "0.101340",
  "Perl::MinimumVersion" => "1.26",
  "Test::More" => "0.88",
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
